array_methods_lab.js

```diff
@@ -0,0 +1,258 @@
+//
+// # Exercises
+//
+// 1. Given an array of numbers, write a function that returns a new array where every element is incremented by 10.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+
+let arr1 = [1, 2, 3, 4, 5]
+
+
+let result = arr1.map (function(currentElem){
+  return currentElem + 10
+})
+
+console.log(result);
+
+// 2. Given an array, return a new array where every non-string element is converted to an empty string.
+//
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+
+let arr2 = ["a", 123, "b", "c", {name: "cat"}];
+let result2 = arr2.filter(elem => {
+  if (typeof(elem) !== 'string') {
+    return ""
+  } else if (typeof(elem === 'string')) {
+    return elem
+  }
+});
+
+console.log(result2);
+
+// 3. Write a function that returns only the even elements in an array of numbers.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+// ```
+let arr3 = [1, 2, 3, 4, 5];
+
+let result3 = arr3.filter(currentElem => {
+  return currentElem % 2 === 0;
+})
+
+console.log(result3);
+
+// 4. Write a function that checks if all elements in the array are the same.
+//
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+//
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+
+let arr4 = [1, 1, 1, 1, 2]
+
+let result4 = arr4.every(elem => {
+  for (let i = 0; i < arr4.length; i++) {
+    return elem === arr4[i]
+  }
+})
+
+console.log(result4);
+
+// 5. Write a function that returns the sum of every element in an array.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+// ```
+
+let arr5 = [1,2,3,4,5];
+let sum = 0
+arr5.forEach(num => sum += num)
+console.log(sum)
+
+// 6. Write a function that finds largest number in an array.
+//
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+// ```
+
+let arr6 = [1,2,6,3,4,5];
+
+let largestNum = arr6.reduce((acc, currentEl) => {
+    if (acc > currentEl) {
+        return acc;
+    } else {
+        return currentEl;
+    }
+});
+
+console.log(largestNum);
+
+// 7. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+//
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+// ```
+let arr7 = [1,0,0,2,0,3,0,0,0];
+
+function countZeroes(arr) {
+  let zero = 0;
+  for(let i = 0; i < arr.length; i++) {
+    if (arr[i] === 0) {
+      zero += 1;
+    }
+  } return zero;
+}
+
```

Comment:
what array method can help you with keeping count as you iterate thought the array?